templates/ImageBuilder/cis-for-eb-image-pipeline-template.yaml

```diff
@@ -0,0 +1,138 @@
+      ImageScanningConfiguration:
+        ImageScanningEnabled: true
+      ImageRecipeArn:
+        Ref: Recipe
+      DistributionConfigurationArn:
+        Ref: ImageDistributionConfiguration
+      ImageTestsConfiguration:
+        TimeoutMinutes: 720
+        ImageTestsEnabled: true
+
+  Recipe:
+    Type: "AWS::ImageBuilder::ImageRecipe"
+    Properties:
+      Name: !Sub ${AWS::StackName}
+      Version: !Ref ImageVersion
+      ParentImage:
+        Fn::Sub: arn:${AWS::Partition}:imagebuilder:${AWS::Region}:aws:image/cis-hardened-image-level-1-on-amazon-linux-2023-arm-r5sqt5uxgygqi/x.x.x
+      Components:
+        - ComponentArn:
+            Ref: CISforEBImageBuilderComponent
+      # The image will not build with the default working dir, /tmp, since CIS makes that directory non-executable
+      WorkingDirectory: "/root"
+      BlockDeviceMappings:
+      - Ebs:
+          Throughput: 125
+          VolumeType: gp3
+          Iops: 3000
+          VolumeSize: !Ref VolumeSize
+          DeleteOnTermination: true
+        DeviceName: "/dev/xvda"
+      AdditionalInstanceConfiguration:
+        SystemsManagerAgent:
+          UninstallAfterBuild: false
+
+  CISforEBImageBuilderComponent:
+    Type: AWS::ImageBuilder::Component
+    Properties:
+      Name: CISforEBImageBuilderComponent
+      Version: 1.0.0
+      Description: Modify CIS Level 1 image to work with Elastic Beanstalk
+      Platform: Linux
+      Data: |
+        name: cis-eb-fixes
+        description: Updates to CIS hardened images to make it work with Elastic Beanstalk
+        schemaVersion: 1.0
+        phases:
+          - name: build
+            steps:
+              - name: xray-group
+                action: ExecuteBash
+                inputs:
+                  commands:
+                    - sudo /usr/sbin/useradd --user-group xray -s /sbin/nologin --no-create-home
+
+  # Specifies the infrastructure within which to build and test your image.
+  # https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-imagebuilder-infrastructureconfiguration.html
+  ImageInfrastructureConfiguration:
+    Type: AWS::ImageBuilder::InfrastructureConfiguration
+    Properties:
+      Name: !Sub '${AWS::StackName}-Configuration'
+      InstanceProfileName:
+        Ref: InstanceProfile
+
+  ImageDistributionConfiguration:
+    Type: AWS::ImageBuilder::DistributionConfiguration
+    Properties:
+      Name: !Sub '${AWS::StackName}-Distributions'
+      Distributions:
+        - Region: !Ref 'AWS::Region'
+          AmiDistributionConfiguration:
+            AmiTags:    # apply tags to generated AMIs
+              Name: !Sub '${AWS::StackName}'
+            LaunchPermissionConfiguration:
+              OrganizationArns:
+                - !Sub 'arn:${AWS::Partition}:organizations::${AWS::AccountId}:organization/${AwsOrganizationId}'
+              UserIds:
+                - 449435941126   # Synapse dev
+                - 325565585839   # Synapse prod
```

Comment:
why not share to all accounts by setting to `- all`?

Reply:
Since the base image is purchased from the AWS Marketplace, setting to 'all' results in an error.

Reply:
hmm, is that limitation documented anywhere?  also did you try `UserGroups: -all` parameter, not `UserIds:`

Reply:
> hmm, is that limitation documented anywhere? 
Not that I've seen.

> also did you try `UserGroups: -all` parameter, not `UserIds:`
Yes, I originally tried following the pattern you used in your Agora template, but it failed, as described above.